Board.java

```diff
@@ -42,11 +42,11 @@ public boolean checkForDraw() {
         return !state.contains("-");
     }
 
-    public boolean checkForWin(char ply) {
+    public boolean checkForWin(char player) {
         for (int[] formation : win) {
             int count = 0;
            for (int i=0; i < formation.length; i++) {
-               if (state.charAt(formation[i]) == ply) count += 1;
+               if (state.charAt(formation[i]) == player) count += 1;
                if (count == 3) return true;
            }
```

Comment:
this is yucky code.. but without using java 8 streams I can't think of anything nicer atm.. maybe you could look into it.